packages/sitecore-jss/src/graphql-request-client.test.ts

```diff
@@ -245,4 +255,154 @@ describe('GraphQLRequestClient', () => {
       expect(client['timeout']).to.equal(300);
     });
   });
+
+  describe('Retrayable status codes', () => {
+    const retryableStatusCodeThrowError = async (statusCode: number) => {
+      nock('http://jssnextweb')
+        .post('/graphql')
+        .reply(statusCode)
+        .post('/graphql')
+        .reply(statusCode)
+        .post('/graphql')
+        .reply(statusCode);
+
+      const graphQLClient = new GraphQLRequestClient(endpoint, {
+        retries: 2,
+      });
+
+      spy.on(graphQLClient['client'], 'request');
+
+      try {
+        await graphQLClient.request('test');
+      } catch (error) {
+        expect(error).to.not.be.undefined;
+        expect(graphQLClient['client'].request).to.have.been.called.exactly(3);
+        spy.restore(graphQLClient);
+      }
+    };
+
+    // Test cases for each retryable status code
+    for (const statusCode of [502, 503, 504, 520, 521, 522, 523, 524]) {
+      it(`should retry and throw error for ${statusCode} when retries specified`, async function() {
+        this.timeout(8000);
+        await retryableStatusCodeThrowError(statusCode);
+      });
+    }
+
+    const retryableStatusCodeResolve = async (statusCode: number) => {
+      nock('http://jssnextweb')
+        .post('/graphql')
+        .reply(statusCode)
+        .post('/graphql')
+        .reply(statusCode)
+        .post('/graphql')
+        .reply(200, {
+          data: {
+            result: 'Hello world...',
+          },
+        });
+
+      const graphQLClient = new GraphQLRequestClient(endpoint, {
+        retries: 3,
+      });
+
+      spy.on(graphQLClient['client'], 'request');
+
+      const data = await graphQLClient.request('test');
+
+      try {
+        await graphQLClient.request('test');
+        expect(data).to.not.be.null;
+      } catch (error) {
+        expect(graphQLClient['client'].request).to.have.been.called.exactly(4);
+        spy.restore(graphQLClient);
+      }
+    };
+
+    // Test cases for each retryable status code
+    for (const statusCode of [429]) {
+      it(`should retry and resolve for ${statusCode} if one of the request resolves`, async function() {
+        this.timeout(16000);
+        await retryableStatusCodeResolve(statusCode);
+      });
+    }
+
+    it('should use custom retryStrategy and retry accordingly', async function() {
```

Comment:
please clarify the test case description based on expected input and output, i.e. should use custom delay and retry strategy when custom strategy provided